core/src/main/java/com/sequenceiq/cloudbreak/init/CloudbreakCleanupService.java

```diff
@@ -119,18 +127,75 @@ private boolean stackToSyncContainsCluster(List<Stack> stacksToSync, Cluster clu
         return stackIds.contains(cluster.getStack().getId());
     }
 
-    private List<Long> restartDistruptedFlows() {
+    /**
+     * Exclude all stacks that have active flows and assigned to Cb nodes.
+     */
+    private List<Long> excludeStacksByFlowAssignment() {
+        List<Long> exclusion = new ArrayList<>();
+        List<Object[]> allNonFinalized = flowLogRepository.findAllNonFinalized();
+        allNonFinalized.stream().filter(o -> o[2] != null).forEach(o -> exclusion.add((Long) o[1]));
```

Comment:
flowLogRepository.findAllNonFinalized() called once in the project, so the filter should be happen in database